Makefile

```diff
@@ -1,8 +1,30 @@
 # Use bash as shell
 SHELL = /bin/bash
 
+# Check for dirty :D 
+define check_dirty
+	GIT_SHA=$$(git rev-parse HEAD) || { \
+		GIT_HASH=$${GIT_SHA:-NO_SHA}; \
+	}; \
+	if [ -z "$$GIT_HASH" ]; then \
+		GIT_DIRTY=$$(git diff --stat); \
+		if [ -n "$$GIT_DIRTY" ]; then \
+			DIRTY="true"; \
+		else \
+			DIRTY="false"; \
+		fi; \
+	fi; \
+	echo $$DIRTY
+endef
+
 # Authorino version
-VERSION = $(shell git rev-parse HEAD)
+AUTHORINO_VERSION = v0.17.3-dev # change as version increments
+
+# Authorino commit hash
+GIT_SHA = $(shell git rev-parse HEAD)
```

Comment:
this is run for every make target, but only used in few of them. Check out https://github.com/Kuadrant/limitador-operator/blob/main/Makefile#L270-L275

Reply:
Absolutely, @eguzki ! Target level implementation is the way to go. :)